stored_requests/backends/db_fetcher/fetcher.go

```diff
@@ -4,30 +4,34 @@ import (
 	"context"
 	"database/sql"
 	"encoding/json"
-
 	"github.com/lib/pq"
 
 	"github.com/golang/glog"
 	"github.com/prebid/prebid-server/stored_requests"
 )
 
-func NewFetcher(db *sql.DB, queryMaker func(int, int) string) stored_requests.AllFetcher {
+func NewFetcher(db *sql.DB, queryMaker func(int, int) string, responseQueryMaker func(int) string) stored_requests.AllFetcher {
```

Comment:
Just for my knowledge, what is happening here in this function signature where I read func(int) in the function signature. Is another function being called in relation to responseQueryMaker in the NewFetcher function signature?

Reply:
Correct. There are two different functions to build queries.
Please refer to the config/stored_requests.go file. 

```
func (cfg *PostgresFetcherQueries) MakeQuery(numReqs int, numImps int) (query string) {
	return resolve(cfg.QueryTemplate, numReqs, numImps)
}

func (cfg *PostgresFetcherQueries) MakeQueryResponses(numIds int) (query string) {
	return resolveQueryResponses(cfg.QueryTemplate, numIds)
}

func resolve(template string, numReqs int, numImps int) (query string) {
	numReqs = ensureNonNegative("Request", numReqs)
	numImps = ensureNonNegative("Imp", numImps)

	query = strings.Replace(template, "%REQUEST_ID_LIST%", makeIdList(0, numReqs), -1)
	query = strings.Replace(query, "%IMP_ID_LIST%", makeIdList(numReqs, numImps), -1)
	return
}

func resolveQueryResponses(template string, numIds int) (query string) {
	numIds = ensureNonNegative("Response", numIds)

	query = strings.Replace(template, "%ID_LIST%", makeIdList(0, numIds), -1)
	return
}

```
There are two different resolvers for db queries. We need them to support stored responses and stored requests db fetch functionality.
For stored requests there is a query that fetches both "Request" and "Imp" in one run using `union` and it has two placeholders: "%REQUEST_ID_LIST%" and "%IMP_ID_LIST%". 
It's complicated and messy to reuse it for responses. 
For responses there is another query that fetches just one type of data in one run, it's simpler and eventually we plan to replace complicated query with `union` to a simple query like we have for responses. Also there is no bundle to data type like `REQUEST_ID_LIST` or `IMP_ID_LIST`. For now I can see this is the cleanest way to support it and make code and configs more or less readable.